src/MIDebugEngine/Engine.Impl/Variables.cs

```diff
@@ -343,23 +343,70 @@ public enum NodeType
 
         private string StripFormatSpecifier(string exp, out string formatSpecifier)
         {
-            formatSpecifier = null;
+            formatSpecifier = null; // will be used with -var-set-format
             int lastComma = exp.LastIndexOf(',');
-            if (lastComma > 0)
-            {
-                string expFS = exp.Substring(lastComma + 1);
-                string trimmed = expFS.Trim();
-                if (trimmed == "x" || trimmed == "X" || trimmed == "h" || trimmed == "H")
-                {
+            if (lastComma <= 0)
+                return exp;
+
+            string expFS = exp.Substring(lastComma + 1);
```

Comment:
need to check length such that the expression doesn't end in a comma. this will add +1 and go beyond the length of `exp`.

Reply:
It will return "" in that case: https://docs.microsoft.com/en-us/dotnet/api/system.string.substring?view=netcore-3.1#System_String_Substring_System_Int32_